client/utils/localStorage.ts

```diff
@@ -0,0 +1,48 @@
+export type LocalStorable = Record<string, any>;
+
+export type LocalStorageWriter<T extends LocalStorable> = {
+	setItem: (item: T) => void;
+	getItem: () => T;
+	getChild: <U extends LocalStorable>(key: string) => LocalStorageWriter<U>;
+};
+
+// See https://en.wikipedia.org/wiki/Private_Use_Areas
+const separator = '\uE000';
+
+const assertPathEntryDoesNotContainSeparator = (entry: string) => {
+	if (entry.includes(separator)) {
+		if (process.env.NODE_ENV === 'production') {
+			throw new Error();
+		} else {
+			throw new Error(
+				`localStorage path entry "${entry}" contains forbidden separator character`,
+			);
+		}
+	}
+};
+
+const createLocalStorageWriter = <T extends LocalStorable>(key: string): LocalStorageWriter<T> => {
+	const setItem = (item: T) => {
+		localStorage.setItem(key, JSON.stringify(item));
+	};
+
+	const getItem = (): T => {
+		return JSON.parse(localStorage.getItem(key) ?? 'null');
```

Comment:
I've never thought about doing it this way (in one line without a if-statement/ternary) before!